capirca/aclgen.py

```diff
@@ -387,70 +388,61 @@ def FilesUpdated(file_name, new_text, binary):
-    new_text = SkipLines(new_text.split('\n'), skip_line_func=p4_tags)
+    conf = filter(p4_tags, conf.split('\n'))
+    new_text = filter(p4_tags, new_text.split('\n'))
 
-  return conf != new_text
+  return list(conf) != list(new_text)
 
 
-def DescendRecursively(input_dirname, output_dirname, definitions, depth=1):
-  """Recursively descend from input_dirname looking for policy files to render.
+def DescendDirectory(input_dirname):
+  """Descend from input_dirname looking for policy files to render.
 
   Args:
     input_dirname: the base directory.
     output_dirname: where to place the rendered files.
     definitions: naming.Naming object.
-    depth: integer, for outputting '---> rendering prod/corp-backbone.jcl'.
 
   Returns:
-    the files that were found
+    a list of dictionaries which include the input file,
+    the output directory, and the definitions object
   """
-  # p4 complains if you try to edit a file like ./corp//corp-isp.jcl
-  input_dirname = input_dirname.rstrip('/')
-  output_dirname = output_dirname.rstrip('/')
-
-  files = []
-  # calling all directories
-  for curdir in [x for x in os.listdir(input_dirname) if
-                 os.path.isdir(input_dirname + '/' + x)]:
-    # be on the lookout for a policy directory
-    if curdir == 'pol':
-      for input_file in [x for x in os.listdir(input_dirname + '/pol')
-                         if x.endswith('.pol')]:
-        files.append({'in_file': os.path.join(input_dirname, 'pol', input_file),
-                      'out_dir': output_dirname,
-                      'defs': definitions})
-    else:
-      # so we don't have a policy directory, we should check if this new
-      # directory has a policy directory
-      if curdir in FLAGS.ignore_directories:
-        continue
-      logging.warning('-' * (2 * depth) + '> %s' % (
-          input_dirname + '/' + curdir))
-      files_found = DescendRecursively(input_dirname + '/' + curdir,
-                                       output_dirname + '/' + curdir,
-                                       definitions, depth + 1)
-      logging.warning('-' * (2 * depth) + '> %s (%d pol files found)' % (
-          input_dirname + '/' + curdir, len(files_found)))
-      files.extend(files_found)
-
-  return files
+  input_dir = pathlib.Path(input_dirname)
+
+  policy_files = []
+  policy_directories = filter(lambda path: path.is_dir(), input_dir.glob('**/pol'))
+  for ignored_directory in FLAGS.ignore_directories:
+    policy_directories = filter(
+      lambda path: not path.match('%s/**/pol' % ignored_directory),
+      policy_directories
+    )
+
+  for directory in policy_directories:
+    directory_policies = list(directory.glob('*.pol'))
+    depth = len(directory.parents) - 1
+    logging.warning(
+      '-' * (2 * depth) + '> %s (%d pol files found)'
+      % (directory, len(directory_policies))
+    )
+    policy_files.extend(filter(lambda path: path.is_file(), directory_policies))
+
+  return policy_files
```

Comment:
this could be done more succinctly by globbing for the `.pol` files instead of `/pol` directories, but that would make the logging output harder to achieve